PukingPredator/Assets/Scripts/Audio/AudioManager.cs

```diff
@@ -0,0 +1,71 @@
+using UnityEngine;
+using System.Collections.Generic;
+using UnityEngine.UIElements;
+
+public class AudioManager : MonoBehaviour
+{
+    [SerializeField]
+    private List<AudioClip> sfxClips;
+
+    [SerializeField]
+    private AudioClip backgroundTrack;
+
+    private AudioSource backgroundSource;
+    private AudioSource sfxSource;
+
+    private Dictionary<string, AudioClip> sfxDictionary = new Dictionary<string, AudioClip>();
+
+    public static AudioManager Instance { get; private set; }
```

Comment:
we already have the SingletonMonobehaviour abstract class to inherit from to achieve singleton and DontDestroyOnLoad behaviours, reference how GameInput uses it